src/Microsoft.DotNet.XHarness.iOS.Shared/Listeners/SimpleTcpListener.cs

```diff
@@ -41,7 +41,10 @@ protected override void Stop()
         {
             _client?.Close();
             _client?.Dispose();
-            _server?.Stop();
+
+            // _server?.Stop(); was causing hangs
+            // https://github.com/dotnet/xharness/issues/73
+            _server?.Server?.Shutdown(SocketShutdown.Both);
```

Comment:
Brute force, but it works :)

Reply:
Yeah, it's not 100% but also no difference to not closing the connection properly either :)